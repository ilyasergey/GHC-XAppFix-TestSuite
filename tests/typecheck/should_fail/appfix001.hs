{-# LANGUAGE ApplicativeFix #-}

module ShouldCompile where

import Control.Applicative
import Control.Monad
import Control.Applicative.Compose
import Control.Applicative.Fix

foo :: ApplicativeFix p => p Bool
foo = alet c = Compose (pure (return True)) in c
