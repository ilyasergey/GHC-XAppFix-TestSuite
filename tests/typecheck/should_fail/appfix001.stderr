
appfix001.hs:11:31:
    Could not deduce (Monad b) arising from a use of `return'
    from the context (ApplicativeFix p)
      bound by the type signature for foo :: ApplicativeFix p => p Bool
      at appfix001.hs:11:1-48
    or from (Applicative b)
      bound by the alet bindings at appfix001.hs:11:7-48
    Possible fix:
      add (Monad b) to the context of
        the alet bindings
        or the type signature for foo :: ApplicativeFix p => p Bool
    In the first argument of `pure', namely `(return True)'
    In the first argument of `Compose', namely `(pure (return True))'
    In the expression: Compose (pure (return True))
